meson.build

```diff
@@ -441,6 +441,30 @@ if get_option('build_backends')
     files += cuda_gen.process(cuda_files_nvcc_fp16, extra_args: ['-arch=compute_70', '-code=sm_70'])
     has_backends = true
   endif
+  
+  ## ~~~~~~~~
+  ## DirectX
+  ## ~~~~~~~~
+  
+  # we should always be able to build DirectX12 backend on windows platform  
+  if host_machine.system() == 'windows' and get_option('dx')
```

Comment:
I'm a bit in doubt whether it's ok that building it with `-Ddx=true` in linux won't fail, but I guess that's fine. :)
Alternative would be to make it a "combo" option with values "true", "false", "auto" (default).

Reply:
We use similar logic in the other backend options, that can be true and just won't be built if the dependencies are not there.